cpu/stm32/periph/gpio_all.c

```diff
@@ -98,31 +99,30 @@ static inline int _pin_num(gpio_t pin)
 static inline void port_init_clock(GPIO_TypeDef *port, gpio_t pin)
 {
     (void)port; /* <-- Only used for when port G requires special handling */
-#if defined(CPU_FAM_STM32F0) || defined (CPU_FAM_STM32F3) || defined(CPU_FAM_STM32L1)
+
+#if defined(RCC_AHBENR_GPIOAEN)
     periph_clk_en(AHB, (RCC_AHBENR_GPIOAEN << _port_num(pin)));
-#elif defined (CPU_FAM_STM32L0) || defined(CPU_FAM_STM32G0) || \
-    defined(CPU_FAM_STM32C0)
-    periph_clk_en(IOP, (RCC_IOPENR_GPIOAEN << _port_num(pin)));
-#elif defined(CPU_FAM_STM32L4) || defined(CPU_FAM_STM32WB) || \
-      defined(CPU_FAM_STM32G4) || defined(CPU_FAM_STM32L5) || \
-      defined(CPU_FAM_STM32U5) || defined (CPU_FAM_STM32WL)
-#if defined(CPU_FAM_STM32U5)
+#elif defined(RCC_AHB1ENR_GPIOAEN)
+    periph_clk_en(AHB1, (RCC_AHB1ENR_GPIOAEN << _port_num(pin)));
+#elif defined(RCC_AHB2ENR_GPIOAEN)
+    periph_clk_en(AHB2, (RCC_AHB2ENR_GPIOAEN << _port_num(pin)));
+#elif defined(RCC_AHB2ENR1_GPIOAEN)
     periph_clk_en(AHB2, (RCC_AHB2ENR1_GPIOAEN << _port_num(pin)));
+#elif defined(RCC_AHB4ENR_GPIOAEN)
+    periph_clk_en(AHB4, (RCC_MC_AHB4ENSETR_GPIOAEN << _port_num(pin)));
```

Comment:
missmatch ??
```suggestion
#elif defined(RCC_MC_AHB4ENSETR_GPIOAEN)
    periph_clk_en(AHB4, (RCC_MC_AHB4ENSETR_GPIOAEN << _port_num(pin)));
```

Reply:
Good catch. Fixed. I am wondering why the build server didn't catch this?

Reply:
Interesting. Building locally (`make -C examples/blinky BOARD=stm32mp157c-dk2`), the mismatch is detected. Does CI/CD build for all boards for PRs.

Reply:
it does not build for all board, but more like one per kind (eg one stm32, one sam)